### PR TITLE
Create API Contract for User Mutation

### LINKED ISSUE #50: Create API Contract for User Mutation on Profile Page/Edit info modal
Mandatory field: id
Acceptable fields: image, description, email, streetAddress, city, state, zip, country

## Sample Mutation Request:

```GraphQL
{
  userUpdate(id: 1, city: "Kenai", state: "Alaska"){
    streetAddress
    city
    state
    country 
  }
}
```

## Sample Response:

```json
{
    "data": {
        "userUpdate": {
            "streetAddress": "123 Admin St",
            "city": "Kenai",
            "state": "Alaska",
            "country": "USA"
        }
    }
}
```

### LINKED ISSUE #53: User Mutation Tests
Develop tests for  User Update Mutations

- [ ] All new attributes tested

- [ ] Sad path/edge case tests included

- [ ] Ensure all tests pass

### PR DESCRIPTION
**Pull request by Jonathan-M-Wilson**
### Description
- Add spec for the user mutation 
- Add migrations to user to add additional attributes (image, description, streetAddress, city, state, zip, country)
- Update seedfile to fill the additional fields we added to user.
- Add testing tools (jest, supertest) 
- Test our new added functionality

### Closes issue(s)
Closes #50 

### Testing
All added/new functionality has been tested and is passing
- visit code/api/src/modules/user/tests for more info
- Remember to run ```npm run test``` instead of ```jest```. 

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [x] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [x] I ran full test suite - all tests passing

### CODE REVIEW

File: code/api/package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "start": "nodemon src/index.js --exec babel-node",
     "start:prod": "npm run build:prod && npm run start:server",
     "build:prod": "babel src -s -D -d build",
-    "start:server": "node build/index.js"
+    "start:server": "node build/index.js",
+    "test": "jest --watch --runInBand --detectOpenHandles"
```
Comment (helloeduardo) [inline code/api/package.json:16]: Thanks for adding this script!

File: code/api/src/modules/user/mutations.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { GraphQLString, GraphQLInt } from 'graphql'
 
 // App Imports
 import { UserType } from './types'
-import { create, remove } from './resolvers'
+import { create, remove, update } from './resolvers'
```
Comment (helloeduardo) [inline code/api/src/modules/user/mutations.js:6]: Nice job importing the new resolver action 😅

**Review by araspaas — APPROVED** [review]
Lookin good to me. great work.

File: code/api/src/modules/user/tests/userMutation.test.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  afterAll(async ()=>{
+      await models.User.destroy({ where: {} })
+  })
+
+  it("updateUser - can update user profile information", async(done) => {
+    let email = 'test email address'
+    let description = 'test description'
+    const response = await request(server)
+      .post('/')
+      .send({ query: `mutation { userUpdate(description: "${description}", email: "${email}") {name email image description streetAddress city state zip country }}`})
```
Comment (helloeduardo) [inline code/api/src/modules/user/tests/userMutation.test.js:61]: This is super useful for me to remember how to test a mutation! Looks good!

Comment (helloeduardo) [reply code/api/src/modules/user/tests/userMutation.test.js:61]: Also, are those backticks necessary, I couldn't recall if they were?

**Review by helloeduardo — APPROVED** [review]
image

**Review by Todd-Estes — APPROVED** [review]

File: code/api/src/modules/user/tests/userMutation.test.js (hunk 3 above)
Comment (helloeduardo) [reply code/api/src/modules/user/tests/userMutation.test.js:61]: AH, STRING INTERPOLATION
